cluster-creation-cli/README.md

```diff
@@ -0,0 +1,353 @@
+
+## Requirements
+
+- Users should be able to **re-use a configuration** they have used for cluster creation previously, optionally overriding only a few details.
+
+- A **dry run option** should allow "previewing" the result without actually writing any resources to the management cluster.
+
+- A **connection with the management cluster** should only be required to write resources. For a dry run, no connection should be necessary. (However, an internet connection will be required to access public information from GitHub repositories.)
+
+- Configuration provided by the user should be **validated** against the apps's schema, and in case of validation errors, users should receive detailed feedback.
+
+- Users should decide whether to actually create the cluster, or print a manifest only.
+
+- A **cluster name must be provided by the user**. We won't generate a cluster name for the user.
+  - At Giant Swarm we decided around November 2023 that we want users to set workload cluster names and avoid auto-generated names ([roadmap#2999](https://github.com/giantswarm/roadmap/issues/2999)). In theory, we could still offer an option (a flag) to auto-generate the name. However, this would require knowledge of the cluster-PROVIDER app schema in the command logic. For example, as of cluster-aws v0.73.0, the cluster name is set in the property .global.metadata.name. Although we are working towards alignment across providers, this may be different in other versions or other provider apps.
+
+## Basic syntax
+
+The command should be named `create cluster`, hence the entire command, without any flags, would be:
+
+    kubectl gs create cluster
+
+The most important flags would be:
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--release`: the version number of the release to use. The release specifies the version of components to use in the cluster. If not specified, the latest release is used.
+- `--namespace` / `-n`: the namespace in which to create the cluster. This is expected to be an organization namespace, like `org-myorg`.
+- `--dry-run`: only validate the configuration and optionally print the resulting manifest.
+- `--output`: This flag can be used to specify the format of the output when `--dry-run` is set. See the "Dry run output" section below for details.
+  - `yaml` for a Kubernetes manifest that includes an App resource and one or several ConfigMap resources.
+  - `config-yaml` for merged configuration YAML.
```

Comment:
What is the difference between an App+ConfigMap manifest vs. "merged configuration YAML"?

Reply:
The `config-yaml` should only include the values part (like `values.yaml`, not a ConfigMap), while the `yaml` would be several manifests for App, ConfigMap. "Merged" means that when using several sources and flags as an input, the output is only one YAML document.

Reply:
As mentioned in the SIG product call: `yaml` vs. `config-yaml` is unclear. I suggest `--output {app,values}`. And we should also reword the RFC here – "merged configuration YAML" should be turned into an unambiguous explanation.